server/asset/eth/rpcclient.go

```diff
@@ -183,92 +182,14 @@ func (c *rpcclient) transaction(ctx context.Context, hash common.Hash) (tx *type
 // accountBalance gets the account balance, including the effects of known
 // unmined transactions.
 func (c *rpcclient) accountBalance(ctx context.Context, assetID uint32, addr common.Address) (*big.Int, error) {
-	tip, err := c.blockNumber(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("blockNumber error: %v", err)
-	}
-
-	// We need to subtract and pending outgoing value, but ignore any pending
-	// incoming value since that can't be spent until mined. So we can't using
-	// PendingBalanceAt or BalanceAt by themselves.
-	// We'll iterate tx pool transactions and subtract any value and fees being
-	// sent from this account. The rpc.Client doesn't expose the
-	// txpool_contentFrom => (*TxPool).ContentFrom RPC method, for whatever
-	// reason, so we'll have to use CallContext and copy the mimic the
-	// internal RPCTransaction type.
-	var txs map[string]map[string]*RPCTransaction
-	err = c.caller.CallContext(ctx, &txs, "txpool_contentFrom", addr)
```

Comment:
I haven't found any publicly available (paid or not) RPC providers that have any methods from the `txpool` namespace available.  I'm thinking we can have server identify which method to take on connect via the `CheckAPIModules` function you added, however I'm not seeing `rpc_modules` supported anywhere either!  Perhaps using a third-party RPC provider with server is gonna be trickier than I'd like.

Reply:
Yeah, when I tested the modules were only available from geth itself.

Reply:
Been stewing on this, and I think we should revive it with a mini project to support an RPC provider that does not provide it, but use the smart balance if it does.

The aim would then be to allow the operator to use a third party RPC provider that they trust, ideally with their account's own API key so rate limits aren't so prevalent.

This is going to be key to make testnet or just mainnet "testing" less of burden.  At present, it's ridiculously difficult, time consuming, and expensive to get geth+prysm running on either networkk.